convokit/model/corpusComponent.py

```diff
@@ -1,49 +1,155 @@
+from random import randrange
+
 from .convoKitMeta import ConvoKitMeta
 from convokit.util import warn, deprecation
-from typing import List, Optional
+from typing import List, Optional, Type
+
+from convokit.storage import DBDocumentMapping, StorageManager
 
 
 class CorpusComponent:
+    def __init__(self,
+                 obj_type: str,
+                 owner=None,
+                 id=None,
+                 vectors: List[str] = None,
+                 meta=None,
+                 storage: Optional[StorageManager] = None,
+                 from_db=False):
+
+        if storage is not None:
+            self.storage = storage
+        elif owner is not None:
+            self.storage = owner.storage
+        else:
+            self.storage = StorageManager('mem')
+            self.storage.setup_collections(None, None, None, None)
+
+        if from_db:
+            return
+
+        if id is None and obj_type != 'conversation':
+            id = randrange(0, 100)
+
+        if ((obj_type == 'utterance' and id in self.storage._utterances)
+                or (obj_type == 'speaker' and id in self.storage._speakers)):
+            id = f'{id}.1'
```

Comment:
Could you explain this part of the design? Code smell...

Reply:
I'm not too sure as Charlotte wrote this. However, I think that if there is duplicates, we add something at the end of id to make it a new one just to prevent overwriting?

Reply:
@calebchiam regarding the line `id = randrange(0,100)` to produce a fresh ID, I agree this is not great. In a different part of the code I flagged a similar line for generating random Corpus IDs and asked Oscar to change it to use the UUID package (the "proper" way to create fresh IDs), which he did, but I must have missed this one. @oscarso2000 can you change this one to generate a UUID as well? (it's highly suspect that the code as-is has a random range of only 100, that's bound to lead to collisions...)

Regarding L34-L36, this is for tracking different versions of the same object (there is similar code in Corpus as well). I think as long as we fix the previous issue with the random IDs, we should be fine to keep this part of the code, as it is highly unlikely that with proper UUID generation you will get collisions unless the user is explicitly trying to update an object with a specified ID

Reply:
I have updated that. and also caught another thing to remove "from random import randrange" since we now use and import uuid instead.